src/SSHDebugPS/RawShell.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using liblinux;
+using liblinux.Shell;
+using Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier;
+
+namespace Microsoft.SSHDebugPS
+{
+    public interface IRawShell : IDisposable
+    {
+        event EventHandler<string> OutputReceived;
+        event EventHandler<int> Closed;
+        event EventHandler ErrorOccured;
+
+        void WriteCommandStart(string startCommand);
+        void Write(string text);
+        void WriteLine(string text);
+    }
+
+    internal class RawLocalShell : IRawShell
+    {
+        private System.Diagnostics.Process _localProcess;
+        private StreamWriter _stdoutWriter;
+
+        public RawLocalShell(string command, string arguments)
+        {
+            ProcessStartInfo processStartInfo = new ProcessStartInfo(command, arguments);
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.RedirectStandardInput = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.CreateNoWindow = true;
+
+            _localProcess = System.Diagnostics.Process.Start(processStartInfo);
+            _stdoutWriter = new StreamWriter(_localProcess.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), 4096, leaveOpen: true);
+            _localProcess.OutputDataReceived += OnProcessOutput;
+            _localProcess.ErrorDataReceived += OnErrorOutput;
+            _localProcess.Exited += OnProcessExited;
+
+            _localProcess.BeginOutputReadLine();
+            _localProcess.BeginErrorReadLine();
+            _localProcess.EnableRaisingEvents = true;
+        }
+
+        public event EventHandler<string> OutputReceived;
+        public event EventHandler<int> Closed;
+        public event EventHandler ErrorOccured;
+
+        public void Dispose()
+        {
+            if (_localProcess != null)
+            {
+                _localProcess.Close();
+            }
+
+            if (_stdoutWriter != null)
+            {
+                _stdoutWriter.Close();
+                _stdoutWriter = null;
+            }
+        }
```

Comment:
I've found it very helpful to have a `DisposableObject` base class that implements `IDisposable` correctly and provides utilities like `EnsureNotDisposed` so you don't have to write that test everywhere.  Richard wrote a good implementation that we're using in PZ if you want it.